### PR TITLE
Improve readme

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
README.md

### Description
This PR fixes a wrong command in the README and rewords the section about the production build.

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

**Review by DerMolly — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ You'll need at least Node 12. We use [yarn](https://yarnpkg.com/) for our depend
 2. Go inside the repo (e.g. `cd hedgedoc-react-client`)
 3. Run `yarn install`
 4. Either run
-    - `yarn start:dev` (expects [a server](https://github.com/hedgedoc/hedgedoc/tree/develop) running under [http://localhost:3000](http://localhost:3000))
-    - `yarn start` (makes all api calls to the same domain the react-client runs on (normally [http://localhost:3001](http://localhost:3001) ))
+    - `yarn start` - Calls only mocked version of the api. Doesn't need a hedgedoc backend.
+    - `yarn start:for-real-backend` - Expects [a hedgedoc backend server](https://github.com/hedgedoc/hedgedoc/tree/develop) running under [http://localhost:3000](http://localhost:3000))
```
Comment (ErikMichelson) [inline README.md:26]:
```suggestion
    - `yarn start:for-real-backend` - Expects [a HedgeDoc backend server](https://github.com/hedgedoc/hedgedoc/tree/develop) running under [http://localhost:3000](http://localhost:3000))
```

File: README.md [hunk 2]
```diff
@@ -22,8 +22,8 @@ You'll need at least Node 12. We use [yarn](https://yarnpkg.com/) for our depend
 2. Go inside the repo (e.g. `cd hedgedoc-react-client`)
 3. Run `yarn install`
 4. Either run
-    - `yarn start:dev` (expects [a server](https://github.com/hedgedoc/hedgedoc/tree/develop) running under [http://localhost:3000](http://localhost:3000))
-    - `yarn start` (makes all api calls to the same domain the react-client runs on (normally [http://localhost:3001](http://localhost:3001) ))
+    - `yarn start` - Calls only mocked version of the api. Doesn't need a hedgedoc backend.
```
Comment (ErikMichelson) [inline README.md:25]:
```suggestion
    - `yarn start` - Calls only mocked version of the api. Doesn't need a HedgeDoc backend.
```

**Review by ErikMichelson — CHANGES_REQUESTED** [review]
Although these change requests are incidentals, I'd like to follow our branding guide (in terms of capitalization) in our own project.

File: README.md (hunk 2 above)
Comment (mrdrogdrog) [reply README.md:25]: done

File: README.md (hunk 1 above)
Comment (mrdrogdrog) [reply README.md]: done

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**